src/canvas.cpp

```diff
@@ -44,6 +45,35 @@ static cairo_surface_t *create_surface(GtkWidget *widget)
 #endif
 }
 
```

Comment:
1. Shouldn't the 3 functions below take a file name string as input, so they're more general purpose?  Having output always write to the same file names may be limiting.

2. What happens if the file can't be created?  Is a NULL pointer returned?  Should be commented.

3. What should the widget correspond to?  I'm not a gtk expert, but if it isn't obvious what this widget is it should be commented.  Can it be any widget?  Is it assumed to be the widget that represents the main graphics / drawing area on the screen?  If so that should be commented.